src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -126,17 +126,26 @@ public function get( $url ) {
 				$headers  = $response->getHeaders();
 				$body     = $response->getBody();
 			} catch ( FailedToGetFromRemoteUrl $exception ) {
-				$status = $exception->getStatusCode();
-				$expiry = new DateTimeImmutable( "+ {$this->min_expiry} seconds" );
-				$body   = $exception->getMessage();
+				$status   = $exception->getStatusCode();
+				$expiry   = new DateTimeImmutable( "+ {$this->min_expiry} seconds" );
+				$body     = $exception->getMessage();
+				$response = new RemoteGetRequestResponse( $body, $headers, $status );
 			}
 
-			$cached_data = new CachedData( $body, $expiry );
+			$cached_data = new CachedData( compact( 'body', 'status', 'headers' ), $expiry );
```

Comment:
Instead of mixing an object with an arbitrary, untyped array, it would be better to adapt the object instead.

How about renaming `CachedData` to `CachedResponse` and replacing the `$value` property with `$body`, `$headers` and `$status` properties instead.

This would also mean changing the transient prefix constant at the top to make sure we skip existing cache entries, perhaps by just adding `_v2` to the prefix.

This way, we can still have strong typing and don't need these `isset()` calls everywhere. We can also provide `isValid()`/`wasSuccessful()` or any other methods to `CachedResponse` that might make usage more straight-forward.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4509/commits/83a14b318d375744169d7d0e5a8551f40666c762.